mindsdb/integrations/handlers/s3_handler/s3_handler.py

```diff
@@ -0,0 +1,246 @@
+from typing import Optional
+from collections import OrderedDict
+
+import pandas as pd
+import boto3
+import io
+import ast
+
+from mindsdb_sql import parse_sql
+from mindsdb_sql.render.sqlalchemy_render import SqlalchemyRender
+from mindsdb.integrations.libs.base_handler import DatabaseHandler
+
+from mindsdb_sql.parser.ast.base import ASTNode
+
+from mindsdb.utilities.log import log
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+
+
+class S3Handler(DatabaseHandler):
+    """
+    This handler handles connection and execution of the S3 statements.
+    """
+
+    name = 's3'
+
+    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+        """
+        Initialize the handler.
+        Args:
+            name (str): name of particular handler instance
+            connection_data (dict): parameters for connecting to the database
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 's3'
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> StatusResponse:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        if self.is_connected is True:
+            return self.connection
+
+        self.connection = boto3.client(
+            's3',
+            aws_access_key_id=self.connection_data['aws_access_key_id'],
+            aws_secret_access_key=self.connection_data['aws_secret_access_key'],
+            region_name=self.connection_data['region_name']
+        )
+        self.is_connected = True
+
+        return self.connection
+
+    def disconnect(self):
+        """ Close any existing connections
+        Should switch self.is_connected.
+        """
+        self.is_connected = False
+        return
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check connection to the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.error(f'Error connecting to Databricks {self.connection_data["schema"]}, {e}!')
```

Comment:
@ZoranPandovski I changed it to the following,
Error connecting to AWS with the given credentials

I hope this is OK, since we are establishing a connection to AWS first using `boto3` before querying the bucket.